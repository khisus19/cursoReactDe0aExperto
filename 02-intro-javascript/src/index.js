

const api_key = "9hNhYfR5jnRYPkOi33k6MxzUwtPNqJFt";

const peticion = fetch(`api.giphy.com/v1/gifs/random?api_key=${ api_key }`);

peticion.then( resp => resp.json() )
  .then( ({ data }) => {
    console.log(data.images.original.url);
  })
  .catch( console.warn );