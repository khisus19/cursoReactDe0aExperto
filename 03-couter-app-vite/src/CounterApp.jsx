import { useState } from 'react';
import PropTypes from "prop-types";


export const CounterApp = ({ value }) => {
  
  const [ counter, setCounter ] = useState(value);

  const handleAdd = () => setCounter( counter + 1);
  const handleSub = () => setCounter( counter - 1);
  const handleReset = () => setCounter( value );

  return (
    <>
      <h1>Counter App</h1>
      <p>{ counter }</p>
      <button onClick={ handleAdd } >+1</button>
      <button onClick={ handleSub } >-1</button>
      <button onClick={ handleReset } >Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number,
}

CounterApp.defaultProps = {
  name: "Jesús Álvarez",
  subtitle: "No hay subtitulo",
  // title: "No hay titulo",
}