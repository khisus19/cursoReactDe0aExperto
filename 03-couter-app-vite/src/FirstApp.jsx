import PropTypes from "prop-types";

import React from 'react'

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  name: "Jesús Álvarez",
  subTitle: "No hay subtitulo",
  // title: "No hay titulo",
}