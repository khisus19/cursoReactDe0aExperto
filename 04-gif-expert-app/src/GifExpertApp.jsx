import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  
  const [categories, setCategories] = useState([ "One punch", "Dragon ball" ]);

  const onAddCategory = () => {
    setCategories([...categories, "Inuyasha"])
  }

  return (
    <>
      {/* titulo */}
      <h1>GifExpertApp</h1>

      {/* Input */}
      <AddCategory />

      {/* Listado de gif */}
      <button onClick={ onAddCategory }>Agregar Categoria</button>
      <ol>
        { categories.map(category => {
          return <li key={category}>{ category }</li>
        }) }
      </ol>
        {/* Gif Item*/}



    </>
  )
}
