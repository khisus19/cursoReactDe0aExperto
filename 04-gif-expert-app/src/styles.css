* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Manrope', sans-serif;
  color: var(--clr-light);
}
:root {
  --clr-main: #00d9f6;
  --clr-dark: #0e121b;
  --clr-grey: #6c7589;
  --clr-light: #fff;
  --ff-secondary: 'Source Sans Pro', sans-serif;
}

/* NAVBAR */
nav {
  width: 100%;
  padding: 20px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  font-weight: 600;
}
#logo img {
  width: 250px;
}
@media (min-width: 600px) {
  nav {
    display: flex;
    flex-direction: row;
  }
}
.nav-link-ul {
  width: 100%;
  max-width: 350px;
  list-style: none;
  font-size: 1rem;
  font-weight: 600;
  display: flex;
  justify-content: space-evenly;
}
li {
  margin: 5px;
}
a, a:visited, a:focus {
  text-decoration: none;
  color: var(--clr-main);
}
a:hover {
  color: var(--clr-grey);
}

/* Body */
body {
  background-color: var(--clr-dark);
}
main {
  padding: 25px;
  display: flex;
  flex-direction: column;
  align-items: center;
}

form {
  margin: 15px auto;
  max-width: 600px;
  width: 90%;
}
input {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
  padding: 10px 15px;
  width: 100%;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin: 5px auto;
  text-align: center;
}

.card-grid {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}

.card {
  padding: 10px;
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  border: 2px double var(--clr-grey);
  display: flex;
  flex-direction: column;
  height: 0%;  
  justify-content: center;
  margin-bottom: 20px;
  margin-right: 20px;
  overflow: hidden;
}

.card p {
  background-color: white;
  color: var(--clr-dark);
  flex: 1;
  font-size: 1.5rem;
  font-weight: 600;
  margin-top: 5px;
  padding: 10px 20px;
  text-align: center;
}

.card img {
  width: 100%;
}