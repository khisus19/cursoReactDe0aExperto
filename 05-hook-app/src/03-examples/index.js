

export * from "./CharacterDisplay";
export * from "./CharacterLoading";