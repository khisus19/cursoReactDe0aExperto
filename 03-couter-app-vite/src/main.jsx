import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from "./FirstApp";
import { App } from './HelloWorldApp'


ReactDOM.createRoot( document.getElementById("root")).render(
  <React.StrictMode>
    <App />
    <FirstApp />
    
  </React.StrictMode>
)