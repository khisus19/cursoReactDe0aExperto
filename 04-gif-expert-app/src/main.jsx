import React from 'react';
import ReactDOM from 'react-dom/client';
import { GifExpertApp } from "./GifExpertApp";
import "./styles.css"
import { NavBar } from "./components/NavBar";


ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <NavBar />
    <main>
      <GifExpertApp />
    </main>
  </React.StrictMode>,
);
