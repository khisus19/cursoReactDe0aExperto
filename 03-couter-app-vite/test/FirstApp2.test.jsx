import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('Pruebas de componente FirstApp', () => { 

  const title = "Hola, Soy Goku";
  const subTitle = "Soy un subtitulo";
  
  test('Debe hacer match con el snapshot', () => { 
    
    const { container } = render( <FirstApp title={ title } /> );
    expect( container ).toMatchSnapshot();
  });

  test('Debe mostrar el mensaje "Hola, Soy Goku', () => { 
    
    render( <FirstApp title={ title } /> );
    expect( screen.getByText(title) ).toBeTruthy();
    // screen.debug();
  })

  test("Debe mostrar un titulo en un h1", () => {
    render( <FirstApp title={ title } /> );
    expect( screen.getByRole("heading", { level: 1 }).innerHTML ).toContain(title)
  });

  test("Debe mostrar el subtitulo", () => {
    render( 
      <FirstApp 
        title={ title }
        subTitle={ subTitle } 
      /> 
    );

    expect( screen.getAllByText(subTitle).length ).toBe(2);

  })
  
});