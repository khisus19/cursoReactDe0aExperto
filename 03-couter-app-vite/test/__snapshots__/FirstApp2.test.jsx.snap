// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de componente FirstApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Goku
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Jesús Álvarez
  </p>
</div>
`;
